app/code/Magento/Customer/Model/Indexer/Source.php

```diff
@@ -61,6 +61,22 @@ public function addFieldToSelect($fieldName, $alias = null)
         return $this;
     }
 
+    /**
+     * Add attribute to entities in collection
+     *
+     * If $attribute == '*' select all attributes
+     *
+     * @param array|string|integer|\Magento\Framework\App\Config\Element $attribute
+     * @param bool|string $joinType flag for joining attribute
+     * @return $this
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function addAttributeToSelect($attribute, $joinType = false)
```

Comment:
Where is this introduced public function used? Also, an introduction of new public methods to existed classes should be avoided to avoid backward compatibility issues.